Source/Client/Sound/CachedSound.cs

```diff
@@ -0,0 +1,31 @@
+using System.Collections.Generic;
+using System.Linq;
+using NAudio.Wave;
+
+namespace FireAndForgetAudioSample
+{
+    class CachedSound
+    {
+        public float[] AudioData { get; private set; }
+        public WaveFormat WaveFormat { get; private set; }
+        public CachedSound(string audioFileName)
+        {
+            using (var audioFileReader = new AudioFileReader(audioFileName))
+            {
+                // TODO: could add resampling in here if required
```

Comment:
Is it required, though?

Reply:
This is a necessary measure, because without it, a format mismatch error is generated.